peers/app_request_network.go

```diff
@@ -257,3 +270,62 @@ func (n *AppRequestNetwork) ConnectPeers(nodeIDs set.Set[ids.NodeID]) set.Set[id
 
 	return trackedNodes
 }
+
+// Helper struct to hold connected validator information
+// Warp Validators sharing the same BLS key may consist of multiple nodes,
+// so we need to track the node ID to validator index mapping
+type ConnectedCanonicalValidators struct {
+	ConnectedWeight       uint64
+	TotalValidatorWeight  uint64
```

Comment:
I think those ideas make sense, but I'm going to defer that for later. I think we'll likely want to add an async routine to update the validator set, rather than fetching it in every message relay, at which point we'll want to carefully consider how the global state is managed. For now though, this type is only used in the message relayer, so the risk of it being misused is fairly low.